sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2548,16 +2548,20 @@ object functions {
   def to_date(e: Column): Column = withExpr { ToDate(e.expr) }
 
   /**
-   * Returns date truncated to the unit specified by the format.
+   * Returns timestamp truncated to the unit specified by the format.
```

Comment:
Yeap. So, I initially wanted to match input type to output type (`DateType` input - `DateType` - output, `TimestampType` input - `TimestampType` output) but just decided to follow the suggestion because It seems it depends on how we define the type maybe as it seems the implementation is different in each DBMS.

I would like to to hear the thoughts on https://github.com/apache/spark/pull/14788#issuecomment-243081134 in more details - cc @cloud-fan